compiler/rustc_trait_selection/src/traits/project.rs

```diff
@@ -1455,7 +1455,12 @@ pub fn compute_inherent_assoc_ty_args<'a, 'b, 'tcx>(
     // Infer the generic parameters of the impl by unifying the
     // impl type with the self type of the projection.
     let self_ty = alias_ty.self_ty();
-    match selcx.infcx.at(&cause, param_env).eq(DefineOpaqueTypes::No, impl_ty, self_ty) {
+    let infcx = selcx.infcx.at(&cause, param_env);
+    let InferOk { value: self_ty, obligations: mut normalize_obligations } =
```

Comment:
Don't use `infcx.normalize` here, use `normalize_with_depth_to` and put it into the same `if !selcx.infcx.next_trait_solver()` block as above.